opm/parser/eclipse/EclipseState/IOConfig/RestartConfig.cpp

```diff
@@ -360,6 +360,24 @@ RPTSCHED( const DeckKeyword& keyword ) {
 }
 
 
+    void RestartConfig::addRestartKeywords( size_t timeStep , const std::map<std::string, int>& input_mnemonics) {
```

Comment:
Externally it's immutable (which is what really matters, honestly), and this requires¹ mutation anyway. You're effectively doing remove-then-add anyway, it's just via conditional addition, preventing whole-structure copying.

Besides, what happens when ALLPROPS is set to 0, and BG is set to 1 later? Which one kicks in?

¹ Not really, but in C++ we can safely operate under this assumption.